http/http_client.hpp

```diff
@@ -106,7 +106,7 @@ struct ConnectionPolicy
 
     size_t maxConnections = 1;
 
-    std::string retryPolicyAction = "TerminateAfterRetries";
+    std::string retryPolicyAction = "RetryForever";
```

Comment:
This looks different than 1030 
https://github.com/ibm-openbmc/bmcweb/blob/1030/http/http_client.hpp#L88

Reply:
From the upstream code - its not applying the retry policy which the user supplies while subscribing for events.  This structure overrides the user's retry policy to TerminateAfterRetries always. 
In 1030, the user supplied retry policy is taken into the http client object while sending the events. Since we have HMC as the subscriber who wants to RetryForever - I modified this for downstream.

Reply:
So we are allowing forcing RetryForever ?

Why can't the HMC set the retryPolicy they want (e.g. RetryForever ) ? 
We are planning on the GUI using events for 1060.. Allowing for event subscriptions from the GUI. I

Reply:
> So we are allowing forcing RetryForever ?

Yes, since HMC needs this. Upstream forces TerminateAfterRetries.

> 
> Why can't the HMC set the retryPolicy they want (e.g. RetryForever ) ? We are planning on the GUI using events for 1060.. Allowing for event subscriptions from the GUI. I worry something like this, against the Redfish spec, is a step in the wrong direction

HMC sets the retry policy as RetryForever. The event subscription is saved and persisted with this value as expected. But the ConnectionPool & http client code at upstream does not take this in. There is something in this agregator design, where they always force TerminateAfterRetries for each https_client. We dont want this to happen for 1050. 

We need to explore this design, to see how to pass the user supplied retry policy to this class object for push style event subscribers - without impacting the aggregator.

Reply:
> There is something in this agregator design, where they always force TerminateAfterRetries for each https_client.

Can we fix this? It seems like a bug, the HMC is setting RetryForever but the bmc isn't actually obeying that

Reply:
> > There is something in this agregator design, where they always force TerminateAfterRetries for each https_client.
> 
> Can we fix this? It seems like a bug, the HMC is setting RetryForever but the bmc isn't actually obeying that

We can try fixing. But that needs proper understanding of the aggregator implementation. Thus, i have made this change for now to solve the downstream event problem. We have several sev1 defects on this.

Reply:
I'm struggling because it feels like we are doing a hack here to make the HMC usecase work but then would break any other client.
And we are working around a real bug

Reply:
Can you open a story to address this correctly? 
For 1060, the GUI was looking to support subscriptions and this change is a move in the opposite direction of that goal